### PR TITLE
Build File Generator: Add BuildFileParser

### PR DESCRIPTION
**Pull request by val-fed**

### CODE REVIEW

File: tools/build_file_generator/BuildFileAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+            result.put(currentAttribute, new ArrayList<>());
+          }
+          String attributeValue = getSubstringByRegex(line, textBetweenDoubleQuotesRegex);
+          if (!attributeValue.isEmpty()) {
+            result
+                .get(currentAttribute)
+                .add(getSubstringByRegex(line, textBetweenDoubleQuotesRegex).replace("\"", ""));
+          }
+          i++;
+        } while (i != ruleContent.size()
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

File: tools/build_file_generator/BuildFileAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+        break;
+      case checkstyle_test:
+        builder.addCheckstyleTest(getCheckstyleTestExtension(getRuleAttributes(ruleContent)));
+        break;
+      default:
+        System.out.println(String.format("%s rule isn't supported.", rule));
+    }
+  }
+
+  // Returns map with attribute name as key and  list of attribute values as the value
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

File: tools/build_file_generator/BuildFileAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+
+public class BuildFileAnalyzer {
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

File: tools/build_file_generator/BuildFileAnalyzer.java [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+            .map(String::trim)
+            .collect(Collectors.toList());
+
+    for (Rule rule : Rule.values()) {
+      for (int i = 0; i < lines.size(); i++) {
+        List<String> ruleContent = new ArrayList<>();
+        if (lines.get(i).startsWith(rule.name())) {
+          do {
+            ruleContent.add(lines.get(++i));
+          } while (!lines.get(i + 1).equals(")"));
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

File: tools/build_file_generator/BuildFileAnalyzer.java [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        Arrays.stream(fileContent.split(System.lineSeparator()))
+            .map(String::trim)
+            .collect(Collectors.toList());
+
+    for (Rule rule : Rule.values()) {
+      for (int i = 0; i < lines.size(); i++) {
+        List<String> ruleContent = new ArrayList<>();
+        if (lines.get(i).startsWith(rule.name())) {
+          do {
+            ruleContent.add(lines.get(++i));
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

File: tools/build_file_generator/BuildFileAnalyzer.java [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+        default:
+          System.out.println(
+              String.format(
+                  "%s attribute isn't supported for checkstyle_test rule.", entry.getKey()));
+      }
+    }
+    return result.build();
+  }
+
+  private enum Rule {
```
[the remark this answers is not in the document]
Comment (val-fed) [reply tools/build_file_generator/BuildFileAnalyzer.java]: Done.

File: tools/build_file_generator/BuildFileParser.java [hunk 7]
```diff
@@ -94,36 +106,29 @@ private void addRule(BuildFile.Builder builder, Rule rule, List<String> ruleCont
...
-          String attributeValue = getSubstringByRegex(line, textBetweenDoubleQuotesRegex);
-          if (!attributeValue.isEmpty()) {
-            result
-                .get(currentAttribute)
-                .add(getSubstringByRegex(line, textBetweenDoubleQuotesRegex).replace("\"", ""));
-          }
-          i++;
-        } while (i != ruleContent.size()
-            && !ruleAttributes.contains(ruleContent.get(i).split(" ")[0]));
+    String currentAttribute = "";
```
Comment (oferb) [inline tools/build_file_generator/BuildFileParser.java]: Great improvement :)

**Pull request merged**